atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -143,6 +148,51 @@ public void timestampServiceShouldRespectDistinctClientsWhenIssuingTimestamps()
     }
 
     @Test
+    public void timestampServiceRespectsTimestampManagementService() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+        TimestampManagementService timestampManagementService = getTimestampManagementService(CLIENT_1);
+
+        long newTimestamp = timestampService.getFreshTimestamp() + ONE_MILLION;
+        timestampManagementService.fastForwardTimestamp(newTimestamp);
+        assertThat(timestampService.getFreshTimestamp()).isGreaterThan(newTimestamp);
+    }
+
+    @Test
+    public void timestampManagementServiceRespectsTimestampService() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+        TimestampManagementService timestampManagementService = getTimestampManagementService(CLIENT_1);
+
+        long newTimestamp = timestampService.getFreshTimestamp() + ONE_MILLION;
+        timestampManagementService.fastForwardTimestamp(newTimestamp);
+        for (int i = 0; i < FORTY_TWO; i++) {
+            timestampService.getFreshTimestamp();
+        }
+        timestampManagementService.fastForwardTimestamp(newTimestamp + 1);
+        assertThat(timestampService.getFreshTimestamp()).isGreaterThan(newTimestamp + FORTY_TWO);
+    }
+
+    @Test
+    public void fastForwardRespectsDistinctClients() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+        TimestampManagementService differentClientTimestampManagementService = getTimestampManagementService(CLIENT_2);
+
+        long currentTimestamp = timestampService.getFreshTimestamp();
+        differentClientTimestampManagementService.fastForwardTimestamp(currentTimestamp + ONE_MILLION);
+        assertEquals(currentTimestamp + 1, timestampService.getFreshTimestamp());
+    }
+
+    @Test
+    public void fastForwardToThePastDoesNothing() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+        TimestampManagementService timestampManagementService = getTimestampManagementService(CLIENT_1);
+
+        long currentTimestamp = timestampService.getFreshTimestamp();
+        timestampManagementService.fastForwardTimestamp(currentTimestamp + TWO_MILLION);
```

Comment:
`currentTimestamp + TWO_MILLION` extract to `currentTimestampIncrementedByTwoMillion`.